### PR TITLE
Document set view

### LINKED ISSUE #77: Document set view
- [x] Add "Document sets" button in the navigation bar.
- [x] Show the list of document sets.
- [x] Show the list of versions for a document set.

### PR DESCRIPTION
**Pull request by The0Jan**
Closes #77

### CODE REVIEW

File: frontend/src/document_set/DocumentsSet.tsx
Comment (MortonPL) [inline frontend/src/document_set/DocumentsSet.tsx:1]: Use a table with sortable columns.

File: frontend/src/set_listed/SetListed.tsx
Comment (MortonPL) [inline frontend/src/set_listed/SetListed.tsx:1]: Use a table with sortable columns.

File: frontend/src/version_set/VersionsSet.tsx
Comment (MortonPL) [inline frontend/src/version_set/VersionsSet.tsx:1]: Use a table with sortable columns.

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -17,6 +17,9 @@ import Versions from './versions/Versions';
 import ApiClient from './api/ApiClient';
 import BackendApiClient from './api/BackendApiClient';
 import VersionEditor from './versions/VersionEditor';
+import DocumentsSet from './document_set/DocumentsSet';
+import VersionsSet from './version_set/VersionsSet';
+import SetListed from './set_listed/SetListed';
```
Comment (MortonPL) [inline frontend/src/App.tsx:22]: Not a big fan of these names.

**Review by MortonPL — DISMISSED** [review]
Comments aside, everything could use an auto-formatter pass.

File: frontend/src/document_set/DocumentsSet.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+};
+
+type DocumentNamedVersionSet = {
+  documentSetName: String,
+  documentVersionSet: DocumentVersionSet
+}
+
+const getFormattedDate = (createdAt: string) => new Date(createdAt).toLocaleString('ro-RO');
+
+export const compareByCreationTime = (first: { createdAt: string }, second: { createdAt: string }) => {
```
[the remark this answers is not in the document]
Comment (Dove6) [reply frontend/src/document_set/DocumentsSet.tsx:20]: Just import from Documents and remove this one.

File: frontend/src/RoutingRoot.tsx [hunk 3]
```diff
@@ -29,7 +29,6 @@ function RoutingRoot({ loginState, apiClient }: RoutingRootProps) {
   (<Badge pill bg={'danger'}>{unreadNotifications}</Badge>) : (<></>)
   const loggedInLinks = (<>
     <Nav.Link as={Link} to="/">Home</Nav.Link>
-    <Nav.Link as={Link} to="/notifications">Notifications {notificationMarker}</Nav.Link>
```
[the remark this answers is not in the document]
Comment (The0Jan) [reply frontend/src/RoutingRoot.tsx:32]: yeah fixed that, i have no idea what happend here

File: frontend/src/set_listed/SetListed.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+          return;
+        apiClient.getVersionSets(documentSetId)
+          .then(response => {             
+            response.forEach(
+                setVersion => {
+                    if(setVersion.setVersionId === versionSetId)
+                    {
+                        setVersion.documentVersionIds.forEach(
+                            (pair,index) => {
+                              console.log(pair);
```
[the remark this answers is not in the document]
Comment (The0Jan) [reply frontend/src/set_listed/SetListed.tsx:37]: yes.

File: frontend/src/set_listed/SetListed.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                                apiClient.getDocument(pair[0]).then(doc => setDocs([...docs,  doc] ));
+                                apiClient.getVersion(pair[0], pair[1]).then(ver => setVers([...vers, ver]))
```
[the remark this answers is not in the document]
Comment (The0Jan) [reply frontend/src/set_listed/SetListed.tsx:39]: done.

File: frontend/src/set_listed/SetListed.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import { FunctionComponent, useState, useEffect } from 'react';
+import { Button, Container, Table } from 'react-bootstrap';
+import { useNavigate } from 'react-router';
+import { useSearchParams } from 'react-router-dom';
+
+import ApiClient from '../api/ApiClient';
+import DocumentVersionSet from '../models/DocumentVersionSet';
```
[the remark this answers is not in the document]
Comment (The0Jan) [reply frontend/src/set_listed/SetListed.tsx:7]: si.

File: frontend/src/version_set/VersionsSet.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    if (documentSetId === undefined)
+      return;
+    apiClient.getVersionSets(documentSetId)
+      .then(response => { setVersionSets(response) });
+  }, [apiClient, documentSetId]);
+
+  function compareVersions(a: DocumentVersionSet, b: DocumentVersionSet): number {
+    // MIND THE MINUS, WE WANT THE NEWEST TO APPEAR ON TOP
+    return -String(a.setVersionName).localeCompare(b.setVersionName, undefined, { numeric: true, sensitivity: 'base' });
+  }
```
[the remark this answers is not in the document]
Comment (The0Jan) [reply frontend/src/version_set/VersionsSet.tsx:34]: done

File: frontend/src/document_set/DocumentsSet.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (The0Jan) [reply frontend/src/document_set/DocumentsSet.tsx]: done

**Pull request merged**